portfolio/src/main/webapp/script.js

```diff
@@ -27,3 +27,75 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+
+/**
+ * Adds my name to the page.
+ */
+function getName() {
+    console.log('Getting Name');
+    // The fetch() function returns a Promise because the request is asynchronous.
+    const responsePromise = fetch('/data');
+    // When the request is complete, pass the response into handleResponse().
+    responsePromise.then(handleNameResponse);
+}   
+
+function handleNameResponse(response) {
+    console.log('Handling the response.');
+
+    // response.text() returns a Promise, because the response is a stream of
+    // content and not a simple variable.
+    const textPromise = response.text();
+
+    // When the response is converted to text, pass the result into the
+    // addQuoteToDom() function.
+    textPromise.then(addQuoteToDom);
+}
+
+/** Adds a random quote to the DOM. */
+function addQuoteToDom(quote) {
+  console.log('Adding Sarah to page:');
+
+  const nameContainer = document.getElementById('name-container');
+  
+  nameContainer.innerText = fetch('/my-data-url')  // sends a request to /my-data-url
```

Comment:
Hmm, it looks like your "quote" from the "/data" Java Servlet is not being used.

You probably want:
```
nameContainer.innerText = quote;
```

I don't see a "/my-data-url" Java Servlet in your DataServelet.java file, so I'm not sure `fetch('/my-data-url')` will return anything.